src/main/scala/com/cloudera/sparkts/models/RegressionARIMA.scala

```diff
@@ -0,0 +1,198 @@
+
+import scala.language.postfixOps
+
+/**
+ * Created by baddar on 1/14/16.
+ * This object contains different methods to
+ */
+
+object RegressionARIMA {
+
+  val COCHRANE_ORCHUTT = 1;
+  val REG_ARMA = 2;
+
+  def fit(ts: linalg.Vector[Double], regressors: DenseMatrix[Double], method: Int, others: Any*): RegressionARIMAModel = {
+    val model: RegressionARIMAModel = method match {
+      case COCHRANE_ORCHUTT => {
+        if (others.length == 0)
+          fitCochraneOrchutt(ts, regressors)
+        else {
+          if (!others(0).isInstanceOf[Int])
+            throw new IllegalArgumentException("Maximum iteration parameter to Cochrane orcutt must be integer")
+          if (others.length > 1)
+            throw new IllegalArgumentException("Number of cochrane orcutt arguments can't exceed 3")
+          val maxIter = others(0).asInstanceOf[Int]
+          fitCochraneOrchutt(ts, regressors, maxIter)
+        }
+      }
+      case _ => {
+        throw new UnsupportedOperationException("Regression ARIMA method : " + method + " not defined.")
+      }
+    }
+    return model;
+  }
+
+  /**
+   * Fit linear regression model with AR(1) errors , for references on cochrane orchutt model:
+   * See [[https://onlinecourses.science.psu.edu/stat501/node/357]]
+   * See : Applied Linear Statistical Models - Fifth Edition - Michael H. Kutner , page 492
+   * The method assumes the time series to have the following model
+   *
+   * Y_t = B.X_t + e_t
+   * e_t = rho*e_t-1+w_t
+   * e_t has autoregressive structure , where w_t is iid ~ N(0,&sigma 2)
+   *
+   * Outline of the method :
+   * 1) OLS Regression for Y (timeseries) over regressors (X)
+   * 2)Apply auto correlation test (Durbin-Watson test) over residuals , to test whether e_t still have auto-regressive structure
+   * 3)if test fails stop , else update update coefficients (B's) accordingly and go back to step 1)
+   *
+   * @param ts : Vector of size N for time series data to create the model for
+   * @param regressors Matrix N X K for the timed values for K regressors over N time points
+   * @param maxIter maximum number of iterations in iterative cochrane-orchutt estimation
+   * @return instance of class [[RegressionARIMAModel]]
+   */
+
+
+  def fitCochraneOrchutt(ts: linalg.Vector[Double], regressors: DenseMatrix[Double], maxIter: Int = 10): RegressionARIMAModel = {
+
+    //parameters check
+    (0 until regressors.cols).map(i => {
+      if (regressors(::, i).length != ts.length) {
+        throw new IllegalArgumentException("regressor at column index = " + i + " has length = " + regressors(::, i).length +
+          "which is not equal to time series length (" + ts.length + ")")
+
+      }
+    })
+    val rhoDiffThr = 0.001
+    val rhos = new Array[Double](maxIter)
+    //Step 1) OLS Multiple Linear Regression
+    val yxOlsMultReg: OLSMultipleLinearRegression = new OLSMultipleLinearRegression()
```

Comment:
Can take out the types from these variables (e.g. `: OLSMultipleLinearRegression`) because it's what type they are from what comes after the equals sign